### PR TITLE
Bug 2313492: [release-4.14] Remove check of object type & return in enqueueStorageClusterRequest

### PR DESCRIPTION
**Pull request by malayparida2000**
The enqueueStorageClusterRequest event handler is used to enqueue a reconcile request for the controller from the watched objects. Although the handle is being used by all the watched objects, but it checks the object type and returns if the object is not of type OCSInitialization. Due to this when any thing happens with the watched objects other than OCSInitialization, the reconcile request is not enqueued and the controller does not reconcile the object.

The watch process on certain resources we had on our code is broken in 4.14. But the issue was being masked due to the side effect of other problem where we were doing continuous reconcile each second.Recently Nitin fixed that Bug so now unnecessary reconciles don't happen anymore,https://github.com/red-hat-storage/ocs-operator/pull/2624.

So, Now the issue was noticed as creation of the Virtual Machine CRD was not triggering a reconcile and thus not creating the special Storageclass that it should have. Upon debugging we found the enqueue function we use on 4.14 just returns if the resource is not OCSInitialization, but we are still using that enqueue function for all the watches we have. Hence a reconcile request never gets queued for the Watched Resources.

The issue is not present on 4.15 onwards as I had unintentionally fixed the bug while working on the Separation of ceph toolbox from ocsinitialization as a part of the Multiple Storagecluster Epic in 4.15, https://github.com/red-hat-storage/ocs-operator/pull/2240.

### Hence this bug requires specific fix only in 4.14.z.

### CODE REVIEW

**Comment by malayparida2000** [comment]
The watch process on certain resources we had on our codebis broken in 4.14. But the issue was being masked due to the side effect of other problem where we were doing continuous reconcile each second.Recently Nitin fixed that Bug so now unnecessary reconciles don't happen anymore,https://github.com/red-hat-storage/ocs-operator/pull/2624. 

So, Now the issue was noticed as creation of the Virtual Machine CRD was not triggering a reconcile and thus not creating the special Storageclass that it should have. Upon debugging we found the enqueue function we use on 4.14 just returns if the resource is not OCSInitialization, but we are still using that enqueue function for all the watches we have. Hence a reconcile request never gets queued for the Watched Resources.

The issue is not present on 4.15 onwards as I had unintentionally fixed the bug while working on the Separation of ceph toolbox from ocsinitialization as a part of the Multiple Storagecluster Epic in 4.15, https://github.com/red-hat-storage/ocs-operator/pull/2240.

### Hence this bug requires specific fix only in 4.14.z.

**Review by iamniting — COMMENTED** [review]
looks good to me will approve once the BZ is approved

File: controllers/storagecluster/storagecluster_controller.go
Comment (nb-ohad) [inline controllers/storagecluster/storagecluster_controller.go:140]: The logic here is trying to be both an enqueue func and a filter func, which means the original author was using the wrong tool for the job.
The intent was, as I read it, NOT to enqueue an event for a storage cluster if the OCS init resource is not at the ready phase. We cannot just remove that logic as there must be a reason the logic was introduced in the first place (maybe it solves a different bug) 

I agree that the checks need to be removed here as this enqueue func is being used in watches other than ocs init watches. But we do need to keep the logic intact for ocs init resource.

A more correct fix will be to introduce a new predicate func that assumes OCSInitialization CR and filter the events if the ocs init CR is not in ready phase. Then add that predicate to the ocs init watch. This will be done on top of the removal of these lines from the above function.

**Review by nb-ohad — CHANGES_REQUESTED** [review]

File: controllers/storagecluster/storagecluster_controller.go
Comment (malayparida2000) [reply controllers/storagecluster/storagecluster_controller.go:140]: Acc to the discussion we had with @iamniting who had initially added the check for ocsinit resource being ready, this check no longer makes sense now, @iamniting can shed more light on the reasoning. Full discussion here https://redhat-internal.slack.com/archives/CJW8YBDR7/p1726739675199359?thread_ts=1726559338.202149&cid=CJW8YBDR7.

Just FYI this code has already been removed in later releases 4.15+.

Comment (nb-ohad) [reply controllers/storagecluster/storagecluster_controller.go:140]: ack

**Review by iamniting — APPROVED** [review]

**Pull request merged**
